graviti/dataset/dataset.py

```diff
@@ -5,20 +5,35 @@
 
 """The implementation of the Graviti Dataset."""
 
-from typing import Optional
+from functools import partial, wraps
+from typing import AbstractSet, Any, Callable, Dict, Iterator, Mapping, Optional
 
 from tensorbay.client.status import Status
-from tensorbay.utility.user import UserMutableMapping
 
+# from graviti.client.catalog import get_catalog
+from graviti.client.data import list_data_details
 from graviti.client.dataset import get_dataset
+from graviti.client.segment import list_segments
 from graviti.dataframe.frame import DataFrame
 from graviti.manager.branch import BranchManager
 from graviti.manager.commit import CommitManager
 from graviti.manager.draft import DraftManager
 from graviti.manager.tag import TagManager
+from graviti.utility.lazy import LazyFactory
 
 
-class Dataset(UserMutableMapping[str, DataFrame]):
+def _getter(func: Callable[..., Any]) -> Callable[..., Any]:
```

Comment:
decorator should use TypeVar to indicate the input type is the same as output type.
ref: https://github.com/linjiX/tensorbay-python-sdk/blob/e03ae156c9db410528146b26054796171d54d1f8/tensorbay/utility/deprecated.py#L45